### ISSUE TITLE
Volume of H1_Tetrahedron_Lagrange1_Gauss1

### ISSUE THREAD

**Issue by mediev** [GEOSX/GEOSX#2154]
Dear GEOSX team,

I got an error that the simulator calculates negative volume for a particular tetrahedron. 

***** ERROR
***** LOCATION: /workspaces/GEOSX/src/coreComponents/mesh/CellElementSubRegion.cpp:397
***** Controlling expression (should be false): m_elementVolume[k] <= 0.0
***** Rank 0: Negative volume for element 0 type 5 in subregion 97_tetrahedra

In debug mode i tracked the mistake that took me to H1_Tetrahedron_Lagrange1_Gauss1::determinantJacobianTransformation
It seems that the calculation of volumes in [H1_Tetrahedron_Lagrange1_Gauss1::determinantJacobianTransformation](https://github.com/GEOSX/GEOSX/blob/42f78953d136191203998c498a66edd59cb001f8/src/coreComponents/finiteElement/elementFormulations/H1_Tetrahedron_Lagrange1_Gauss1.hpp#L297) assumes a sequence of points sorted in a way that will provide a non-negative triple product (or determinant). 
For the set of points
[-0.1275    , -0.12517408, -0.107     ],
[-0.13992335, -0.13254333, -0.107     ],
[-0.14015945, -0.115     , -0.11670727],
[-0.14048467, -0.12287681, -0.107     ]
this function produces negative volume which is wrong.

![image](https://user-images.githubusercontent.com/981487/200311350-28cd1d3e-d135-4fdb-ad94-5e8d3737f3d7.png)

Why you don't take the absolute value of the determinant you calculate in this function?

I use a mesh generated by GMSH and exported to *.vtu file that is included in Geosx input file. By the way, this sequence of points for this particular element is the same as in the GMSH output. It is not possible to share *.vtu files here. I shared input files and mesh in the folder in Google Drive you can access by the following link:
https://drive.google.com/drive/folders/1UpxZNGnh6aO11wBf0EULBC-I9YWHNk66?usp=share_link
To run use the following command: ./path/to/geosx -i displaced_reservoir.xml

I use
- Docker with Ubuntu 20.04-gcc9, version 200-813
- GEOSX Version: commit 0fc8b366191c2e2a01b5a359ba6b71718c72a5a7

Best Regards,
Aleks Novikov,
TU Delft

**Comment by TotoGaz** [comment]
Hello @mediev and thank you for your interest in GEOSX

I opened your mesh in `paraview` and I noticed that a portion of it has indeed negative volume.
You can also check using the `MeshQuality` filter.

<img width="735" alt="image" src="https://user-images.githubusercontent.com/49004943/200374027-835c4c77-8138-4228-b53f-14026dd3ff12.png">

There's a whole corner of the mesh with negative volumes, and it seems like GEOSX computes the volumes appropriately.

In a nutshell, the philosophy of GEOSX is to take the input mesh as is and run the simulation on it.
Therefore, it's important that the user checks the mesh validity first.

To help the user in this journey, we are also developing a tool [mesh_doctor](https://github.com/GEOSX/GEOSX/tree/develop/src/coreComponents/python/modules/geosx_mesh_doctor) which is still very young and would not have detected your case actually (even though it's something we would like to be able to propose).
If you think it's valuable and you have time for it, please consider sending us a PR!

Thx!

**Comment by castelletto1** [comment]
Tetrahedra in the region with negative volume are not properly ordered according to the VTK convention. For example, cell 7155 is currently defined by points `{7377, 7384, 7380, 10841}`. However, VTK would expects it to be ordered as `{7377, 7380, 7384, 10841}` (see [here](https://raw.githubusercontent.com/Kitware/vtk-examples/gh-pages/src/Testing/Baseline/Cxx/GeometricObjects/TestLinearCellDemo.png) ), hence the negative volume).

<img width="461" alt="Screen Shot 2022-11-07 at 10 09 28 AM" src="https://user-images.githubusercontent.com/38361926/200384760-6aaec36a-9545-4ff8-bd1f-96cb14c3d6a2.png">

<img width="154" alt="Screen Shot 2022-11-07 at 10 09 37 AM" src="https://user-images.githubusercontent.com/38361926/200384844-3259fa6e-e9c4-4d52-9bcb-dc8db540fdca.png">

**Comment by mediev** [comment]
Now I see the point, it is discussed e.g. here
https://discourse.vtk.org/t/vtkhexahedron-node-ordering-question/4111/3

I constructed the mesh in Gmsh which has the same convention.
https://gmsh.info/doc/texinfo/gmsh.html#Node-ordering

However, I used MSH format version 2.2 to export mesh. And I had this problem with hexahedrons as well. Now I tried MSH 4.1 and this problem disappears in both tetrahedral and hexahedral meshes.

Thank you!
